test/augmentation/test_augmentation.py

```diff
@@ -24,6 +25,794 @@
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=False)
+        output = augmentation(input_tensor.to(self.device).to(self.dtype))
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+
+    def _test_random_p_1_return_transform_implementation(
+            self, input_tensor, expected_output, expected_transformation, params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        output, transformation = augmentation(input_tensor.to(self.device).to(self.dtype))
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+
+    def _test_module_implementation(self, params):
+        augmentation = self._create_augmentation_from_params(**params, p=0.5, return_transform=True)
+
+        augmentation_sequence = nn.Sequential(augmentation, augmentation)
+
+        input_tensor = torch.rand(3, 5, 5, device=self.device, dtype=self.dtype)  # 3 x 5 x 5
+
+        torch.manual_seed(42)
+        out1, transform1 = augmentation(input_tensor)
+        out2, transform2 = augmentation(out1)
+        transform = transform2 @ transform1
+
+        torch.manual_seed(42)
+        out_sequence, transform_sequence = augmentation_sequence(input_tensor)
+
+        assert out2.shape == out_sequence.shape
+        assert transform.shape == transform_sequence.shape
+        assert_allclose(out2, out_sequence)
+        assert_allclose(transform, transform_sequence)
+
+    def _test_inverse_coordinate_check_implementation(self, params):
+        torch.manual_seed(42)
+
+        input_tensor = torch.zeros((1, 3, 50, 100), device=self.device, dtype=self.dtype)
+        input_tensor[:, :, 20:30, 40:60] = 1.
+
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        output, transform = augmentation(input_tensor)
+
+        if (transform == kornia.eye_like(3, transform)).all():
+            pytest.skip("Test not relevant for intensity augmentations.")
+
+        indices = kornia.create_meshgrid(height=output.shape[-2],
+                                         width=output.shape[-1],
+                                         normalized_coordinates=False,
+                                         device=self.device)
+        output_indices = indices.reshape((1, -1, 2)).to(dtype=self.dtype)
+        input_indices = kornia.geometry.transform_points(transform.to(self.dtype).inverse(), output_indices)
```

Comment:
@shijianjian why is not dtype inferred ?

Reply:
I found we have lots of legacy code having a bad device and dtype conversion during the run. We need to have proper testing and cuda CI here.

Might need to make the device and dtype param required for all the testing (they are optional currently).